nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-resources/src/main/resources/conf/nifi.properties

```diff
@@ -244,6 +244,9 @@ nifi.security.user.saml.http.client.read.timeout=${nifi.security.user.saml.http.
 # nifi.security.group.mapping.value.anygroup=$1
 # nifi.security.group.mapping.transform.anygroup=LOWER
 
+# listener bootstrap properties
+nifi.listener.bootstrap.port=${nifi.listener.bootstrap.port}
```

Comment:
My understanding is that they must be different.

Bootstrap will start and listen on `nifi.bootstrap.listen.port`. It will then start the main NiFi process and pass an argument that tells that main NiFi process what port Bootstrap is listening on.

Then when NiFi starts, it will start listening on `nifi.listener.bootstrap.port`, and send a message to Bootstrap (to the port bootstrap is listening on, i.e. `nifi.bootstrap.listen.port`) telling it what this port it started listening on. At this point, each process is listening on a different port and knows which port the other is listening on to send commands.

By default these ports are random ephemeral ports, hence this communication needed to tell each other what the ports the other is listening on. This patch just makes it so you can optionally specify non-random ports.

Reply:
Ah I conflated the names `nifi.bootstrap.listen.port` and `nifi.listener.bootstrap.port`, my bad

Reply:
Maybe more detail in the comment about what the port property means/does?

Reply:
Yeah, the names are a bit confusing, I'm open to other suggestions that might be more clear.

I've added a comment as suggested.